ludwig/features/bag_feature.py

```diff
@@ -94,16 +88,14 @@ def add_feature_data(
 
 @register_input_feature(BAG)
 class BagInputFeature(BagFeatureMixin, InputFeature):
-    encoder = "embed"
-    vocab = []
-
-    def __init__(self, feature, encoder_obj=None):
-        super().__init__(feature)
-        self.overwrite_defaults(feature)
+    def __init__(self, input_feature_config: Union[BagInputFeatureConfig, Dict], encoder_obj=None, **kwargs):
+        input_feature_config = self.load_config(input_feature_config)
+        super().__init__(input_feature_config, **kwargs)
+        self.encoder_config = input_feature_config.encoder
         if encoder_obj:
             self.encoder_obj = encoder_obj
         else:
-            self.encoder_obj = self.initialize_encoder(feature)
+            self.encoder_obj = self.initialize_encoder()
```

Comment:
I'm not a fan of setting the encoder_config to self and then inside self.initialize_encoder() referring to it through self. I would rather pass the encoder_config explicitly to the initialize_encoder function.

Reply:
This applies to basically every input feature type

Reply:
I'm actually working on this as we speak. Changing this to a method where we pass the encoder config into the encoder initializer to create the encoder. The encoder config will then live on the encoder object. This will avoid any issues with tied weights pointing to one encoder object but an encoder config pointing to another.

Reply:
Just finished this switch so we no longer use self in the initialization of encoders and decoders. The config is directly piped in and then becomes an attribute of the `encoder_obj` / `decoder_obj`